New functions: alm algebraic operations and indexing

LeeoBianchi opened the pull request.

ziotom78 commented on docs/src/alm.md:
````diff
@@ -118,16 +112,75 @@ pixwin
 
-You can use the function [`almxfl`](@ref) (or [`almxfl!`](@ref)) to multiply (in-place)
- a set of $a_{\ell m}$ coefficients by an $\ell$-dependent generic function $f_\ell$.
+Depending on which order the arguments are placed, the result will be overwritten
+in place or returned as a new instance of `Alm` type:
+
+```julia
+using Healpix
+
+#just two constant Alm objects
+myalm = Healpix.Alm(5,5, ones(ComplexF64, Healpix.numberOfAlms(5)))
+myf_l = ones(Healpix.numberOfAlms(5)) .* 2
+
+#will return a new object:
+myalm*myf_l
+myalm/myf_l
+
+#will overwrite myalm:
+myf_l*myalm
+myf_l\myalm #here the inverse division must be used
+```
+
+In either case the call to such operator consists in a shortcut to the function [`almxfl`](@ref) (or [`almxfl!`](@ref), if in place).
````
This is counter-intuitive and hard to remember; better to remove in-place operators and force people to use explicit functions instead. (This is also suggested in the [Julia forum](https://discourse.julialang.org/t/overload-addition-assignment-operator/28836/13?u=maurizio_tomasi).)

LeeoBianchi commented:
@ziotom78 I've pushed a couple of simple functions I had to code up for my work which I think could be useful for a Healpix user in general.

With this I think the PR can be merged, if you're happy with it.

Btw, having all these new functions included in a new Healpix.jl version would make things much easier for the release of my new package :)

ziotom78 commented:
> @ziotom78 I've pushed a couple of simple functions I had to code up for my work which I think could be useful for a Healpix user in general.
> 
> With this I think the PR can be merged, if you're happy with it.
> 
> Btw, having all these new functions included in a new Healpix.jl version would make things much easier for the release of my new package :)

Great! I'm fine with these changes, so we can merge the PR. But can you please add a entry in the CHANGELOG listing the functions that this PR adds?

LeeoBianchi commented:
> Great! I'm fine with these changes, so we can merge the PR. But can you please add a entry in the CHANGELOG listing the functions that this PR adds?

I added one last function to get the pixels corresponding to a specific ring in a map, which I had in my scripts and thought could be useful when operating with HealpixMaps. I also updated the CHANGELOG.md.

Everything should be in place now!
